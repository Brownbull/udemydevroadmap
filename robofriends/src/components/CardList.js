import React from 'react'
import Card from './Card'

const CardList = ({ robots }) => {
  console.log('CardList')
  return (
    <div >
      {
        robots.map((element, i) => {
          return (
            <Card
              name={element.name}
              email={element.email}
              key={i}
              id={element.id} />
          )
        })
      }
    </div>
  )
} 

export default CardList
