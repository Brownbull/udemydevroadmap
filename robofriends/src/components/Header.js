// Splitting 1
import React, { Component } from 'react'
// Splitting 6
import CounterButton from './CounterButton'

class Header extends Component {
  // Splitting 4
  shouldComponentUpdate(nextProps, nextState) {
    return false
  }

  render() {
    console.log('Header')
    return (
      <div>
        <h1 className="f1">Robofriends</h1>
        <CounterButton color={'red'} />
      </div>
    )
  } // eof render
}

export default Header