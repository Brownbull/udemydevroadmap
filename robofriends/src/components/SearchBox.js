import React from 'react'

const SearchBox = ({ changeSearch}) => {
  console.log('SearchBox')
  return (
    <div className='tc pa2'>
      <input
        onChange={changeSearch}
        className='pa3 ba b--green bg-lightest-blue'
        type='search' 
        placeholder='search robots'/>
    </div>
  )
}

export default SearchBox
