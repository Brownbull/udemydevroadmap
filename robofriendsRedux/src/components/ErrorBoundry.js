import React, { Component } from 'react'

class ErrorBoundry extends Component {
  constructor(props) {
    super(props)
    this.state = {
      hasError: false
    } // eof this.state
  } // eof constructor

  componentDidCatch(error, info) {
    this.setState({ hasError: true })
  }

  render() {
    if (this.state.hasError){
      return <h1>Error boundry Catch!</h1>
    }
    return this.props.children
  }
} // eof 


export default ErrorBoundry
