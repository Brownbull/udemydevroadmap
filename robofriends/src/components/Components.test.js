import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })
// ReactTest 2
import React from 'react'
import { shallow, mount, render } from 'enzyme'

import Card from './Card'
import CardList from './CardList'
import CounterButton from './CounterButton'
import ErrorBoundry from './ErrorBoundry'
import Header from './Header'
import Scroll from './Scroll'
import SearchBox from './SearchBox'

// it('expect to render Card component', () =>{
//   expect(shallow(<Card />).length).toEqual(1)
// })
// let wrapper

// beforeEach(() => {
//   const mockProps = {
//     onRequestRobots: jest.fn(),
//     robots: [],
//     searchField: '',
//     isPendinf: false
//   } 
//   wrapper = shallow(<MainPage { ...mockProps }/>)
// })

// ReactTest 3
it('Card component Snapshot', () => {
  // Snapshot will take a photo of the current component 
  // and from here compare it against future versions
  // To update snapshot just change ocmponent and use option u on test menu
  expect(shallow(<Card />)).toMatchSnapshot()
})

// ReactTest 4
it('CardList component Snapshot', () => {
  const MockRobots = [
    {
      id: 99,
      name: 'John Snow',
      email: 'john@goth.com'
    }
  ]
  expect(shallow(<CardList robots={MockRobots} />)).toMatchSnapshot()
})

it('SearchBox component Snapshot', () => {
  expect(shallow(<SearchBox />)).toMatchSnapshot()
})

it('Scroll component Snapshot', () => {
  expect(shallow(<Scroll />)).toMatchSnapshot()
})

// ReactTest 5
it('CounterButton component Snapshot: Change Color', () => {
  const mockColor = 'red'
  expect(shallow(<CounterButton color={mockColor} />)).toMatchSnapshot()
})

it('CounterButton component Snapshot: Increment counter', () => {
  const mockColor = 'red'
  const wrapper = shallow(<CounterButton color={mockColor} />)
  wrapper.find('[id="counter"]').simulate('click')
  wrapper.find('[id="counter"]').simulate('click')
  expect(wrapper.state()).toEqual({ count: 2 })
  wrapper.find('[id="counter"]').simulate('keypress')
  expect(wrapper.state()).toEqual({ count: 2 })
  expect(wrapper.props().color).toEqual('red')
})

it('ErrorBoundry component Snapshot', () => {
  expect(shallow(<ErrorBoundry />)).toMatchSnapshot()
})

it('Header component Snapshot', () => {
  expect(shallow(<Header />)).toMatchSnapshot()
})

