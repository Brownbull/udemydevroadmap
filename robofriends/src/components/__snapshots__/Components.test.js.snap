// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="robot"
          src="https://robohash.org/undefined?size=200x200"
/>,
        <div>
          <h2 />
          <p />
</div>,
      ],
      "className": "tc bg-light-green pa2 dib br3 ma2 grow bw2 shadow-5",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "robot",
          "src": "https://robohash.org/undefined?size=200x200",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2 />,
            <p />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="robot"
            src="https://robohash.org/undefined?size=200x200"
/>,
          <div>
            <h2 />
            <p />
</div>,
        ],
        "className": "tc bg-light-green pa2 dib br3 ma2 grow bw2 shadow-5",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "robot",
            "src": "https://robohash.org/undefined?size=200x200",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2 />,
              <p />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`CardList component Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList
    robots={
        Array [
            Object {
              "email": "john@goth.com",
              "id": 99,
              "name": "John Snow",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          email="john@goth.com"
          id={99}
          name="John Snow"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "email": "john@goth.com",
          "id": 99,
          "name": "John Snow",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            email="john@goth.com"
            id={99}
            name="John Snow"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "email": "john@goth.com",
            "id": 99,
            "name": "John Snow",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`CounterButton component Snapshot: Change Color 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CounterButton
    color="red"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Count: ",
        0,
      ],
      "color": "red",
      "id": "counter",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      "Count: ",
      0,
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Count: ",
          0,
        ],
        "color": "red",
        "id": "counter",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        "Count: ",
        0,
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`ErrorBoundry component Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorBoundry />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): undefined,
  Symbol(enzyme.__nodes__): Array [
    undefined,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Header component Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="f1"
>
          Robofriends
</h1>,
        <CounterButton
          color="red"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Robofriends",
          "className": "f1",
        },
        "ref": null,
        "rendered": "Robofriends",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "color": "red",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="f1"
>
            Robofriends
</h1>,
          <CounterButton
            color="red"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Robofriends",
            "className": "f1",
          },
          "ref": null,
          "rendered": "Robofriends",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "color": "red",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Scroll component Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Scroll />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "style": Object {
        "border": "0px solid black",
        "height": "800px",
        "overflowY": "scroll",
      },
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "style": Object {
          "border": "0px solid black",
          "height": "800px",
          "overflowY": "scroll",
        },
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`SearchBox component Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBox />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        aria-label="Search Robots"
        className="pa3 ba b--green bg-lightest-blue"
        onChange={undefined}
        placeholder="search robots"
        type="search"
/>,
      "className": "tc pa2",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-label": "Search Robots",
        "className": "pa3 ba b--green bg-lightest-blue",
        "onChange": undefined,
        "placeholder": "search robots",
        "type": "search",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          aria-label="Search Robots"
          className="pa3 ba b--green bg-lightest-blue"
          onChange={undefined}
          placeholder="search robots"
          type="search"
/>,
        "className": "tc pa2",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-label": "Search Robots",
          "className": "pa3 ba b--green bg-lightest-blue",
          "onChange": undefined,
          "placeholder": "search robots",
          "type": "search",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
